google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2574,9 +2743,16 @@ enum Position {
   @GuardedBy("lock")
   private final Queue<WaiterFuture> waiters = new LinkedList<>();
 
+  @GuardedBy("lock")
+  private final Queue<MultiplexedSessionWaiterFuture> multiplexedSessionWaiters =
```

Comment:
I think that we can simplify and remove this whole waiter if we change multiplexed session(s) from being held in a queue to be held in a `SettableFuture`. Waiting for the multiplexed session would then be reduced to simply calling `Future.get()`, which then naturally blocks until ready, and then returns either the result or the error that happened.